### ISSUE TITLE
Idea: Make TraceEvent class work with DLR

### ISSUE THREAD

**Issue by jazzdelightsme** [microsoft/perfview#617]
TraceEvent could be made more convenient to work with by integrating with the DLR (Dynamic Language Runtime). 

That sounds like a Big Deal, but it just means making TraceEvent derive from [DynamicObject](https://docs.microsoft.com/en-us/dotnet/api/system.dynamic.dynamicobject?view=netframework-4.7.1), or alternatively implementing the [IDynamicMetaObjectProvider](https://docs.microsoft.com/en-us/dotnet/api/system.dynamic.idynamicmetaobjectprovider?view=netframework-4.7.1) interface. (See also: [this PDF](http://download.microsoft.com/download/5/4/B/54B83DFE-D7AA-4155-9687-B0CF58FF65D7/implementing-dynamic-interfaces.pdf))

That would allow changing code like this:

```csharp
TraceEvent te = GetEventSomehow();
object lockName = te.PayloadByName( "LockName" );
```
Into code like this:
```csharp
dynamic te = GetEventSomehow();
object lockName = te.LockName;
```

Additionally, it would allow similarly convenient syntax in other environments that work with the DLR--the one I'm really interested in is PowerShell (@HeathS):

```powershell
# Get the set of all lock names
$edgeCP.EventsInProcess | `
    where ProviderName -eq DManipLockTracer | `
    %{ $PSItem.LockName } | `
    Select -Unique
```

(the DLR interfaces are available in NetStandard2, but not 1.6, which I why I recently suggested the upgrade in the other issue)

I think this would be possible by just overriding two methods (if we went the derive-from-DynamicObject route):

```csharp
public override IEnumerable<string> GetDynamicMemberNames()
{
    return PayloadNames;
}

public override bool TryGetMember( GetMemberBinder binder, out object result )
{
    result = this.PayloadByName( binder.Name );
    return result != null;
}
```
I think this fancy awesomeness is pay-for-play: other than possibly referencing a DLR assembly or something, I don't think it would cause any sort of overhead unless/until DLR machinery is invoked (e.g. casting to `dynamic` in C#).

What do you think?

**Comment by vancem** [comment]
I was initially worried about the dependency on DynamicObject, but it looks pervasive enough that I don't have an objection.  

It sounds like it does not require much code change at all.  If that holds up, I would be OK with merging it.  I would suggest you give it a try and if it stays simple create a pull request and I am open to merging it.

**Comment by brianrob** [comment]
Fixed via #630.
